### PR TITLE
Issue #13501: Kill mutation for AtclauseOrder

### PR DESCRIPTION
**Pull request by Kevin222004**
Issue #13501: Kill mutation for AtclauseOrder

--------

# Check 
https://checkstyle.org/checks/javadoc/atclauseorder.html#AtclauseOrder

--------

# Mutation 
https://github.com/checkstyle/checkstyle/blob/7e879f3eedd9302623a6b2ac383b22af1cbe99f0/config/pitest-suppressions/pitest-javadoc-suppressions.xml#L66-L73

--------

# Explaination 
The tags go trimmed at https://github.com/checkstyle/checkstyle/blob/7e879f3eedd9302623a6b2ac383b22af1cbe99f0/src/main/java/com/puppycrawl/tools/checkstyle/AbstractAutomaticBean.java#L368-L371
before setToken

-----------

# Regression

**Overview**
no diff

Report-1 :- https://checkstyle-diff-reports.s3.us-east-2.amazonaws.com/c9ee9e2_2023120752/reports/diff/index.html

Report-2 :- https://checkstyle-diff-reports.s3.us-east-2.amazonaws.com/c9ee9e2_2023180229/reports/diff/index.html

-------

Diff Regression config: https://gist.githubusercontent.com/Kevin222004/b55841e291e59cfec63ffb63dc635840/raw/5ca37f4703a35b84dfc4e5fcaf52dcc95b11c710/Atclause.xml
Diff Regression projects: https://gist.githubusercontent.com/Kevin222004/9600f179b602d4c971bdb0a050099005/raw/360a95ed7bb60d7a0956e531199d484c4d6f6617/test-projects.properties
Report label: Report-2

### CODE REVIEW

**Comment by Kevin222004** [comment]
Github, generate report

**Comment by Kevin222004** [comment]
Github, generate report

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/AtclauseOrderCheck.java [hunk 1]
```diff
@@ -208,9 +208,7 @@ public void setTarget(String... targets) {
      */
     public void setTagOrder(String... orders) {
         final List<String> customOrder = new ArrayList<>(orders.length);
-        for (String order : orders) {
-            customOrder.add(order.trim());
-        }
+        customOrder.addAll(Arrays.asList(orders));
         tagOrder = customOrder;
```
Comment (romani) [inline src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/AtclauseOrderCheck.java:212]: can it be simply `tagOrder = Arrays.asList(orders);` ?

**Review by romani — CHANGES_REQUESTED** [review]
pitest killing is correct.

small item:

**Comment by romani** [comment]
CI is failing

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/AtclauseOrderCheck.java (hunk 1 above)
Comment (Kevin222004) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/AtclauseOrderCheck.java]: Done

**Review by romani — APPROVED** [review]
Ok to merge

**Review by rdiachenko — APPROVED** [review]
lgtm

**Review by Vyom-Yadav — APPROVED** [review]
LGTM!

**Pull request merged**